openmls/src/framing/validation.rs

```diff
@@ -220,32 +256,25 @@ impl UnverifiedContextMessage {
             }
         }
         match plaintext.sender().sender_type {
-            // We want to return a credential when the sender type is member
-            SenderType::Member => {
-                if let Some(credential) = credential_option {
-                    Ok(UnverifiedContextMessage::Member(UnverifiedMemberMessage {
-                        plaintext,
-                        credential,
-                    }))
-                } else {
-                    // If the sender is a member, there must be a credential
-                    Err(ValidationError::LibraryError)
-                }
+            SenderType::Member | SenderType::NewMember => {
+                Ok(UnverifiedContextMessage::Group(UnverifiedGroupMessage {
+                    plaintext,
+                    credential,
+                }))
             }
-            // TODO #192: We don't support external senders yet
+            // TODO #151: We don't support preconfigured senders yet
```

Comment:
I think the actual issue we want to tag here is #106. 151 is about ReInit specifically.